src/main/java/com/github/charlyb01/music_control/mixin/MusicTrackerMixin.java

```diff
@@ -23,97 +25,158 @@
-                if (MusicControlClient.previousMusic) {
-                    MusicControlClient.previousMusic = false;
-                    Identifier music = MusicCategories.PLAYED_MUSICS.peekLast();
-                    if (music != null) {
-                        MusicControlClient.currentMusic = music;
-                    }
-                } else if (MusicControlClient.musicSelected != null) {
-                    MusicControlClient.currentMusic = MusicControlClient.musicSelected;
-                    MusicControlClient.musicSelected = null;
-                } else if (type != null
-                        && MusicControlClient.currentCategory.equals(Music.DEFAULT_MUSICS)
-                        && Music.MUSIC_BY_EVENT.containsKey(type.getSound().value().getId())) {
-                    ArrayList<Music> musics = new ArrayList<>(Music.MUSIC_BY_EVENT.get(type.getSound().value().getId()));
-                    if (musics.isEmpty()) {
-                        musics = new ArrayList<>(Music.MUSIC_BY_EVENT.get(MusicType.GAME.getSound().value().getId()));
-                    }
-                    MusicControlClient.currentMusic = MusicCategories.getRandomMusicIdentifier(musics, this.random);
-                } else {
-                    MusicControlClient.currentMusic = MusicCategories.getMusicIdentifier(this.random);
-                }
+        // stop current music
+        this.client.getSoundManager().stop(this.current);
+
+        if (MusicControlClient.musicSelected != null) {
+            // a new music is selected from the menu
+            MusicControlClient.currentMusic = MusicControlClient.musicSelected;
+            MusicControlClient.musicSelected = null;
+        } else if (MusicControlClient.previousMusic) {
+            // previous music is assigned
+            MusicControlClient.previousMusic = false;
+            Identifier music = MusicCategories.PLAYED_MUSICS.peekLast();
+            if (music != null) {
+                MusicControlClient.currentMusic = music;
             }
-
-            this.current = PositionedSoundInstance.music(SoundEvent.of(MusicControlClient.currentMusic));
-            if (this.current.getSound() != SoundManager.MISSING_SOUND) {
-                this.client.getSoundManager().play(this.current);
+        } else if (MusicControlClient.loopMusic) {
+            // loop mode is on
+            // do nothing, use the same music
+        } else if (id != null
+                && MusicControlClient.currentCategory.equals(Music.DEFAULT_MUSICS)
+                && Music.MUSIC_BY_EVENT.containsKey(id)) {
+            // normal procedure
+            final var musics = new ArrayList<Music>(Music.MUSIC_BY_EVENT.get(id));
+            if (musics.isEmpty()) {
+                // this means the current event corresponds to
+                // an event with no music (e.g. warped forest).
+                MusicControlClient.currentMusic = new Identifier(EMPTY_MUSIC);
```

Comment:
That's not a bad idea because some may want to keep events empty on purpose, maybe to make some place a bit more creepy without music. But this shouldn't be the default behavior, add a boolean config to let people choose if they want this behavior or if they don't have to worry about empty events which should fallback on GAME event